superset/security/api.py

```diff
@@ -167,29 +167,28 @@ def guest_token(self) -> Response:
     @expose("/custom_login/", methods=["GET"])
     @event_logger.log_this
     @safe
+    @protect()
     @permission_name("read")
     def custom_login(self) -> Response:
         token_val = request.headers.get("Authorization")
 
         headers = {"Authorization": token_val, "accept": "application/json"}
 
         response = requests.get(
-            f"{current_app.config.get('PLATFORM_RBAC_BASE_URL')}/v1/users/me?expand=organizations.accounts",
+            f"{current_app.config.get('PLATFORM_RBAC_BASE_URL')}/v1/users/me",
             headers=headers,
         )
 
         if response.status_code != 200:
-            return self.response(400)
+            return self.response(401)
 
         email = response.json().get("email")
-
-        if not email:
-            return self.response(400)
-
+        dashboard_id = request.args.get("dashboard_id")
         user = security_manager.find_user(email=email)
 
-        login_user(user, remember=False)
+        if not email or not dashboard_id or not user:
+            return self.response(401)
 
-        dashboard_id = request.args.get("dashboard_id")
+        login_user(user, remember=False)
 
         return redirect(f"{request.url_root}/{dashboard_id}")
```

Comment:
The `redirect` is incomplete:

```
        return redirect(
            f"{request.url_root}superset/dashboard/{dashboard_id}/", Response=Response
        )
```

`Response=Response` is required by `mypy` when this change is moved to `superset-kh`.

Reply:
done